frontend/src/static/js/v_charts.js

```diff
@@ -0,0 +1,234 @@
+    },
+  },
+  methods: {
+    getFileTypeContributionBars(fileTypeContribution) {
+      let currentBarWidth = 0;
+      const fullBarWidth = 100;
+      const contributionPerFullBar = (this.avgContributionSize * 2);
+      const allFileTypesContributionBars = {};
+
+      Object.keys(fileTypeContribution).forEach((fileType) => {
+        const contribution = fileTypeContribution[fileType];
+        let barWidth = (contribution / contributionPerFullBar) * fullBarWidth;
+        const contributionBars = [];
+
+        // if contribution bar for file type is able to fit on the current line
+        if (currentBarWidth + barWidth < fullBarWidth) {
+          contributionBars.push(barWidth);
+          currentBarWidth += barWidth;
+        } else {
+          // take up all the space left on the current line
+          contributionBars.push(fullBarWidth - currentBarWidth);
+          barWidth -= fullBarWidth - currentBarWidth;
+          // additional bar width will start on a new line
+          const numOfFullBars = Math.floor(barWidth / fullBarWidth);
+          for (let i = 0; i < numOfFullBars; i += 1) {
+            contributionBars.push(fullBarWidth);
+          }
+          const remainingBarWidth = barWidth % fullBarWidth;
+          if (remainingBarWidth !== 0) {
+            contributionBars.push(remainingBarWidth);
+          }
+          currentBarWidth = remainingBarWidth;
+        }
+        allFileTypesContributionBars[fileType] = contributionBars;
+      });
+
+      return allFileTypesContributionBars;
+    },
+
+    getFileTypes(repo) {
+      const fileTypes = [];
+      repo.forEach((user) => {
+        Object.keys(user.fileTypeContribution).forEach((fileType) => {
+          if (!fileTypes.includes(fileType)) {
+            fileTypes.push(fileType);
+          }
+        });
+      });
+      return fileTypes;
+    },
+
+    getContributionBars(totalContribution) {
+      const res = [];
+      const contributionLimit = (this.avgContributionSize * 2);
+
+      const cnt = parseInt(totalContribution / contributionLimit, 10);
+      for (let cntId = 0; cntId < cnt; cntId += 1) {
+        res.push(100);
+      }
+
+      const last = (totalContribution % contributionLimit) / contributionLimit;
+      if (last !== 0) {
+        res.push(last * 100);
+      }
+
+      return res;
+    },
+
+    getAuthorProfileLink(userName) {
+      return `https://github.com/${userName}`;
+    },
+
+    getRepoLink(repo) {
+      const { REPOS } = window;
+      const { location, branch } = REPOS[repo.repoId];
+
+      if (Object.prototype.hasOwnProperty.call(location, 'organization')) {
+        return `${window.BASE_URL}/${location.organization}/${location.repoName}/tree/${branch}`;
+      }
+
+      return repo.location;
+    },
+
+    // triggering opening of tabs //
+    openTabAuthorship(user, repo, index) {
+      const { minDate, maxDate } = this;
+
+      this.$parent.$emit('view-authorship', {
```

Comment:
Not so ideal, but still acceptable.

Reply:
We might need to use a event bus or Vuex soon to manage the events. Especially for pr #1101 now we try to open zoom from ramp. I am not very familiar with vue event management, if there is other way to circumvent this?

Reply:
Vuex is one option. There is indeed event bus plugin for vue. But I don't think there is a necessity to do so. Maybe we may consider to use vuex first?

Reply:
can, I will open another PR for that